src/main/java/com/google/cloud/flink/bigquery/ProtobufUtils.java

```diff
@@ -0,0 +1,717 @@
+import org.apache.flink.table.types.logical.utils.LogicalTypeDefaultVisitor;
+import org.apache.flink.types.Row;
+
+/**
+ * Utilities for converting Flink logical types, such as convert it to the related TypeSerializer or
+ * ProtoType.
+ */
+@Internal
+public final class ProtobufUtils {
+  private static final String MAPTYPE_ERROR_MESSAGE = "MapType is unsupported.";
+  private static final int MAX_BIGQUERY_NESTED_DEPTH = 15;
+  private static final String EMPTY_STRING = "";
+  private static final String RESERVED_NESTED_TYPE_NAME = "STRUCT";
+
+  private static final ImmutableMap<Field.Mode, DescriptorProtos.FieldDescriptorProto.Label>
+      BigQueryModeToProtoFieldLabel =
+          new ImmutableMap.Builder<Field.Mode, DescriptorProtos.FieldDescriptorProto.Label>()
+              .put(Field.Mode.NULLABLE, DescriptorProtos.FieldDescriptorProto.Label.LABEL_OPTIONAL)
+              .put(Field.Mode.REPEATED, DescriptorProtos.FieldDescriptorProto.Label.LABEL_REPEATED)
+              .put(Field.Mode.REQUIRED, DescriptorProtos.FieldDescriptorProto.Label.LABEL_REQUIRED)
+              .build();
+
+  private static final ImmutableMap<LegacySQLTypeName, DescriptorProtos.FieldDescriptorProto.Type>
+      BigQueryToProtoType =
+          new ImmutableMap.Builder<LegacySQLTypeName, DescriptorProtos.FieldDescriptorProto.Type>()
+              .put(LegacySQLTypeName.BYTES, DescriptorProtos.FieldDescriptorProto.Type.TYPE_BYTES)
+              .put(LegacySQLTypeName.INTEGER, DescriptorProtos.FieldDescriptorProto.Type.TYPE_INT64)
+              .put(LegacySQLTypeName.BOOLEAN, DescriptorProtos.FieldDescriptorProto.Type.TYPE_BOOL)
+              .put(LegacySQLTypeName.FLOAT, DescriptorProtos.FieldDescriptorProto.Type.TYPE_DOUBLE)
+              .put(
+                  LegacySQLTypeName.NUMERIC, DescriptorProtos.FieldDescriptorProto.Type.TYPE_STRING)
+              .put(
+                  LegacySQLTypeName.BIGNUMERIC,
+                  DescriptorProtos.FieldDescriptorProto.Type.TYPE_INT64)
+              .put(LegacySQLTypeName.STRING, DescriptorProtos.FieldDescriptorProto.Type.TYPE_STRING)
+              .put(
+                  LegacySQLTypeName.TIMESTAMP,
+                  DescriptorProtos.FieldDescriptorProto.Type.TYPE_STRING)
+              .put(LegacySQLTypeName.DATE, DescriptorProtos.FieldDescriptorProto.Type.TYPE_INT64)
+              .put(
+                  LegacySQLTypeName.DATETIME,
+                  DescriptorProtos.FieldDescriptorProto.Type.TYPE_STRING)
+              .put(
+                  LegacySQLTypeName.GEOGRAPHY,
+                  DescriptorProtos.FieldDescriptorProto.Type.TYPE_BYTES)
+              .put(LegacySQLTypeName.TIME, DescriptorProtos.FieldDescriptorProto.Type.TYPE_STRING)
+              .put(LegacySQLTypeName.RECORD, DescriptorProtos.FieldDescriptorProto.Type.TYPE_STRING)
+              .build();
+
+  public static FlinkFnApi.Schema.FieldType toProtoType(LogicalType logicalType) {
```

Comment:
Why do we have this method? I do not see it being used in PR #21 

If we are not using it, let us please remove it

Reply:
Removed methods which are not in scope at the moment